libs/ui-lib/lib/cim/components/common/CpuArchitectureDropdown.tsx

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import { Dropdown, DropdownItem, DropdownToggle, FormGroup } from '@patternfly/react-core';
+import { useTranslation } from '../../../common/hooks/use-translation-wrapper';
+import {
+  architectureData,
+  ClusterDetailsValues,
+  CpuArchitecture,
+  getFieldId,
+  StaticField,
+  SupportedCpuArchitecture,
+} from '../../../common';
+import { useField, useFormikContext } from 'formik';
+
+const CpuArchitectureDropdown = ({ isDisabled = false }: { isDisabled?: boolean }) => {
+  const { t } = useTranslation();
+  const { setFieldValue } = useFormikContext<ClusterDetailsValues>();
+  const [{ name, value }, , { setValue }] = useField<SupportedCpuArchitecture>('cpuArchitecture');
+  const [cpuArchOpen, setCpuArchOpen] = React.useState(false);
+  const fieldId = getFieldId(name, 'input');
+
+  const onCpuArchToggle = React.useCallback(() => {
+    if (cpuArchOpen) {
+      setCpuArchOpen(false);
+    } else {
+      setCpuArchOpen(true);
+    }
+  }, [cpuArchOpen]);
```

Comment:
There is no need to memoize this function. It could also be simplified to `setCpuArchOpen(!cpuArchOpen)`